lib/bundler/fetcher/compact_index.rb

```diff
@@ -44,6 +44,8 @@ def specs_for_names(gem_names)
           complete_gems.concat(deps.map(&:first)).uniq!
           remaining_gems = next_gems - complete_gems
         end
+        @bundle_worker.stop if @bundle_worker
+        @bundle_worker = nil # reset it, just incase.  Not sure if necessary
```

Comment:
Okay, just found the test that explains that in `spec/quality_spec.rb` 👍 

To be fair, I was also getting test failures like this locally:

```
  1) bundle executable when ENV['BUNDLE_GEMFILE'] is set to an empty string ignores it
     Failure/Error: run! "require '#{name}.rb'; puts #{version_const}", *groups

     RuntimeError:
       Invoking run!("require 'rack.rb'; puts RACK", {}) failed:
       /Users/nicklamuro/code/public_forks/bundler/lib/bundler/resolver.rb:365:in `block in verify_gemfile_dependencies_are_found!': Could not find gem 'rack' in any of the gem sources listed in your Gemfile or available on this machine. (Bundler::GemNotFound)
        from /Users/nicklamuro/code/public_forks/bundler/lib/bundler/resolver.rb:340:in `each'
        from /Users/nicklamuro/code/public_forks/bundler/lib/bundler/resolver.rb:340:in `verify_gemfile_dependencies_are_found!'
        from /Users/nicklamuro/code/public_forks/bundler/lib/bundler/resolver.rb:198:in `start'
        from /Users/nicklamuro/code/public_forks/bundler/lib/bundler/resolver.rb:181:in `resolve'
        from /Users/nicklamuro/code/public_forks/bundler/lib/bundler/definition.rb:243:in `resolve'
        from /Users/nicklamuro/code/public_forks/bundler/lib/bundler/definition.rb:167:in `specs'
        from /Users/nicklamuro/code/public_forks/bundler/lib/bundler/definition.rb:226:in `specs_for'
        from /Users/nicklamuro/code/public_forks/bundler/lib/bundler/definition.rb:215:in `requested_specs'
        from /Users/nicklamuro/code/public_forks/bundler/lib/bundler/environment.rb:19:in `requested_specs'
        from /Users/nicklamuro/code/public_forks/bundler/lib/bundler/runtime.rb:14:in `setup'
        from /Users/nicklamuro/code/public_forks/bundler/lib/bundler.rb:100:in `setup'
        from -e:1:in `<main>'
     # ./spec/support/matchers.rb:60:in `block in should_be_installed'
     # ./spec/support/matchers.rb:57:in `each'
     # ./spec/support/matchers.rb:57:in `should_be_installed'
     # ./spec/bundler/cli_spec.rb:38:in `block (3 levels) in <top (required)>'
```

But maybe my computer was just as tired as I am... or was hallucinating...